cli/src/utils.rs

```diff
@@ -0,0 +1,135 @@
+            ADAPTERS.to_vec(),
+            vec![],
+        );
+
+        Self {
+            adapters: adapters.iter().map(|&p| p.into()).collect::<Vec<Adapter>>(),
+        }
+    }
+
+    pub fn selections(&self) -> Vec<usize> {
+        self.adapters
+            .iter()
+            .map(|p| p.as_usize())
+            .collect::<Vec<usize>>()
+    }
+
+    pub fn contains(&self, adapter: Adapter) -> bool {
+        self.adapters.iter().any(|p| *p == adapter)
+    }
+}
+
+pub struct PlatformSelector {
+    pub platforms: Vec<Platform>,
+}
+
+impl PlatformSelector {
+    pub fn construct(selections: Vec<String>) -> Self {
+        let mut platforms: Vec<Platform> = vec![];
+        for s in selections {
+            platforms.push(s.as_str().into());
+        }
+        Self { platforms }
+    }
+
+    pub fn select(config: &Config) -> Self {
+        // defaults based on previous selections.
+        let defaults: Vec<bool> = PLATFORMS
+            .iter()
+            .map(|&platform| config.target_platforms.contains(platform))
+            .collect();
+
+        let platforms = multi_select(
+            "Select platform(s) to build for (multiple selection with space)",
+            "No platforms selected. Please select at least one platform.",
+            PLATFORMS.to_vec(),
+            defaults,
+        );
+
+        Self {
+            platforms: platforms
+                .iter()
+                .map(|&p| p.into())
+                .collect::<Vec<Platform>>(),
+        }
+    }
+
+    pub fn eq(&self, platforms: &Vec<Platform>) -> bool {
+        self.platforms.eq(platforms)
+    }
+
+    pub fn contains(&self, platform: Platform) -> bool {
+        self.platforms.iter().any(|p| *p == platform)
+    }
+
+    pub fn select_archs(&self) -> HashMap<String, Vec<String>> {
+        let mut archs: HashMap<String, Vec<String>> = HashMap::new();
+        self.platforms.iter().for_each(|&p| match p {
+            Platform::Ios => {
+                let sel = Self::select_multi_archs(p.into(), &IOS_ARCHS);
+                let sel_str = sel
+                    .iter()
+                    .map(|&i| IOS_ARCHS[i].to_string())
+                    .collect::<Vec<String>>();
+                archs.insert("iOS".to_string(), sel_str);
+            }
+            Platform::Android => {
+                let sel = Self::select_multi_archs(p.into(), &ANDROID_ARCHS);
+                let sel_str = sel
+                    .iter()
+                    .map(|&i| ANDROID_ARCHS[i].to_string())
+                    .collect::<Vec<String>>();
+                archs.insert("Android".to_string(), sel_str);
+            }
+            Platform::Web => {}
+        });
+        archs
+    }
+
+    fn select_multi_archs(platform: &str, archs: &[&str]) -> Vec<usize> {
+        // At least one architecture must be selected
+        multi_select(
+            "Select {} architecture(s) to compile (default: all)",
```

Comment:
should be a format here
```rs
            format!(
                "Select {} architecture(s) to compile (default: all)",
                platform
            )
```

and now it is not defaulted by all 
please check
https://github.com/zkmopro/mopro/blob/dfb9b286c63f6b418fe27465796c818996558bf7/cli/src/build.rs#L257

Reply:
fixed in [6704478](https://github.com/zkmopro/mopro/pull/295/commits/67044788976cbbdc6cfad34a296d82d2a22e726a)

Reply:
Still seeing like "? Select {} architecture(s) to compile "

Reply:
@KimiWu123 Please check this 🙏🏻
```rs
            format!(
                "Select {} architecture(s) to compile (default: all)",
                platform
            )
```

Reply:
Sorry, I misunderstood it. It should be fixed in [28fb131](https://github.com/zkmopro/mopro/pull/295/commits/28fb131c9fdcad986239fa8ddc99f0164ccf9466). 🙏